README.md

```diff
@@ -37,38 +41,46 @@ dbc.connect(uri, mongoClient);
 
 ### Using DI
 
-```javascript
-import { MongoRepository, Collection, Pre, Post } from 'mongtype';
-import { Injectable } from 'injection-js';
+```typescript
+import {MongoRepository, Collection, Pre, Post} from 'mongtype';
+import {Injectable} from 'injection-js';
+import {ObjectID} from "mongodb";
 
-interface User {
-  name: string;
+interface UserDTO {
+    name: string;
+}
+
+interface UserDocument extends UserDTO {
+    id: string | ObjectID;
 }
 
 @Injectable()
 @Collection({
-  name: 'user',
-  capped: true,
-  size: 10000
+    name: 'user',
+    capped: true,
+    size: 10000
 })
-export class UserRepository extends MongoRepository<User> {
-  @Before('save')
-  doSomethingBeforeSave() {}
-
-  @After('save')
-  doSomethingAfterSave() { }
+export class UserRepository extends MongoRepository<UserDocument, UserDTO> {
+    @After('create')
+    doSomethingAfterCreate() {
+    }
+
+    @Before('save')
+    doSomethingBeforeSave() {
+    }
 }
 
 @Injectable()
 export class App {
-  constructor(private userRepo: UserRepository) { }
-
-  async findUsers() {
-    const one = await this.userRepo.findById('3434-34-34343-3434');
-    const many = await this.userRepo.find({ conditions: { name: 'foo' } });
-    const newOne = await this.userRepo.create({ foo: true });
-    const updated = await this.userRepo.save(newOne);
-  }
+    constructor(private userRepo: UserRepository) {
+    }
+
+    async findUsers() {
+        const one = await this.userRepo.findById('3434-34-34343-3434');
+        const many = await this.userRepo.find({conditions: {name: 'foo'}});
+        const newOne = await this.userRepo.create({foo: true});
+        const updated = await this.userRepo.save(newOne);
+    }
```

Comment:
Minor observation, but the indentation in this file is inconsistent with the others. Feel free to ignore.

Reply:
Nah, good catch. Will revise when I have time to fully set up my IDE for this repo.

Reply:
Nah, good catch. I'll revise when I have a moment to more fully set up my IDE for the project.